frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -145,3 +150,82 @@ export const useEventSourceList = (namespace: string) => {
   };
   return eventSourceList;
 };
+
+export const getEventSourceYaml = (namespace: string, eventSource: string): K8sResourceKind => {
+  let eventSourceModel: K8sKind;
+  const name = `${eventSource}`;
+  switch (eventSource) {
+    case 'camelsource':
+      eventSourceModel = EventSourceCamelModel;
+      break;
+    default:
+      eventSourceModel = {
+        apiGroup: KNATIVE_EVENT_SOURCE_APIGROUP_DEP,
+        apiVersion: 'v1alpha1',
+        kind: '',
+        label: '',
+        labelPlural: '',
+        plural: '',
+        abbr: '',
+      };
+  }
+  return {
+    apiVersion: `${eventSourceModel.apiGroup}/${eventSourceModel.apiVersion}`,
+    kind: eventSourceModel.kind,
+    metadata: {
+      name,
+      namespace,
+    },
+    spec: {
+      sink: {
+        ref: {
+          apiVersion: `${ServiceModel.apiGroup}/${ServiceModel.apiVersion}`,
+          kind: ServiceModel.kind,
+          name: '',
+        },
+      },
+    },
+  };
+};
+
+export const getEventSourcesResourceFromYAML = (formData: any): K8sResourceKind => {
+  const {
+    application: { name: applicationName },
+    yamlData,
+  } = formData;
+  const eventYaml = safeLoad(yamlData);
+  const { name } = eventYaml.metadata;
+  const defaultLabel = getAppLabels(name, applicationName);
+  const { apiVersion, kind, spec, metadata } = eventYaml;
+  const eventSourceResource: K8sResourceKind = {
+    apiVersion,
+    kind,
+    metadata: {
+      ...metadata,
+      labels: {
+        ...defaultLabel,
+      },
+      annotations,
+    },
+    spec: {
+      ...spec,
+    },
+  };
+  return eventSourceResource;
+};
```

Comment:
can avoid it

Reply:
PTAL https://github.com/openshift/console/pull/4960#discussion_r407273463